cmd/juju/commands/main.go

```diff
@@ -152,7 +152,7 @@ func (m main) Run(args []string) int {
 	jujuArgs := set.NewStrings(args[1:]...)
 	helpArgs := set.NewStrings("help", "-h", "--help")
 	showHelp := jujuArgs.Intersection(helpArgs).Size() > 0 && jujuArgs.Size() == 1
-	repl := jujuArgs.Size() == 0
+	repl := jujuArgs.Size() == 0 && isTerminal(ctx.Stdout) && isTerminal(ctx.Stdin)
```

Comment:
Why do we need to constrain stdin/out to being a terminal? You might be interested in piping a scripted session from a file or using expect to interact with the repl.

Reply:
I thought about that but decided to be overly cautious instead. The reason being that if you want to script something, then script it the "normal" way. Happy to just check stdout if you feel strongly about it.

Reply:
No particular preference and as you said, you can still script it anyway. Just curious as to why it was constrained.